ee/clickhouse/queries/breakdown_props.py

```diff
@@ -7,104 +7,77 @@
     prop_filters, prop_filter_params = parse_prop_clauses(
         filter.properties,
         team_id,
-        table_name=_event_table_name,
+        table_name="e",
+        prepend="e_brkdwn",
         filter_test_accounts=filter.filter_test_accounts,
         person_properties_column="person_props",
         allow_denormalized_props=True,
     )
 
     entity_params, entity_format_params = get_entity_filtering_params(
-        entity, team_id, table_name=_event_table_name, with_prop_filters=True, person_properties_column="person_props"
+        entity, team_id, with_prop_filters=True, person_properties_column="person_props", table_name="e"
     )
 
-    value_expression, _ = get_property_string_expr("person", cast(str, filter.breakdown), "%(key)s", "person_props")
-
-    elements_query = TOP_PERSON_PROPS_ARRAY_OF_KEY_SQL.format(
-        value_expression=value_expression,
-        parsed_date_from=parsed_date_from,
-        parsed_date_to=parsed_date_to,
-        person_query=ClickhousePersonQuery(filter, team_id).get_query(),
-        prop_filters=prop_filters,
-        aggregate_operation=aggregate_operation,
-        GET_TEAM_PERSON_DISTINCT_IDS=GET_TEAM_PERSON_DISTINCT_IDS,
-        **entity_format_params,
-    )
-    top_elements_array = _get_top_elements(
-        filter=filter,
-        team_id=team_id,
-        query=elements_query,
-        params={**prop_filter_params, **entity_params, **extra_params, "key": filter.breakdown},
-        limit=limit,
-    )
-
-    return top_elements_array
-
-
-def get_breakdown_event_prop_values(
-    filter: Filter, entity: Entity, aggregate_operation: str, team_id: int, limit: int = 25, extra_params={}
-):
-    parsed_date_from, parsed_date_to, _ = parse_timestamps(filter=filter, team_id=team_id)
-    prop_filters, prop_filter_params = parse_prop_clauses(
-        filter.properties,
-        team_id,
-        table_name="e",
-        prepend="e_brkdwn",
-        filter_test_accounts=filter.filter_test_accounts,
-    )
-
-    entity_params, entity_format_params = get_entity_filtering_params(entity, team_id, with_prop_filters=True)
+    if filter.breakdown_type == "person":
+        value_expression, _ = get_property_string_expr("person", cast(str, filter.breakdown), "%(key)s", "person_props")
+    else:
+        value_expression, _ = get_property_string_expr("events", cast(str, filter.breakdown), "%(key)s", "properties")
 
-    value_expression, _ = get_property_string_expr("events", cast(str, filter.breakdown), "%(key)s", "properties")
+    person_join_clauses = ""
+    person_query = ClickhousePersonQuery(filter, team_id, column_optimizer=column_optimizer)
+    if person_query.is_used:
+        person_join_clauses = f"""
+            INNER JOIN ({GET_TEAM_PERSON_DISTINCT_IDS}) AS pdi ON e.distinct_id = pdi.distinct_id
+            INNER JOIN ({person_query.get_query()}) person ON pdi.person_id = person.id
```

Comment:
Would it be possible/preferable to just join on person here rather than a subquery?

Reply:
It would not be because we need to handle deleted rows in-query rather than at-table level. See more in ee/clickhouse/queries/person_query.py